Sources/SwiftIfConfig/IfConfigEvaluation.swift

```diff
@@ -0,0 +1,516 @@
+        let stringLiteral = argExpr.as(StringLiteralExprSyntax.self),
+        stringLiteral.segments.count == 1,
+        let segment = stringLiteral.segments.first,
+        case .stringSegment(let stringSegment) = segment
+      else {
+        // FIXME: better diagnostic here
+        throw IfConfigError.unknownExpression(condition)
+      }
+
+      let versionString = stringSegment.content.text
+      let expectedVersion = try VersionTuple(parsingCompilerBuildVersion: versionString, argExpr)
+
+      guard let actualVersion = configuration.compilerVersion else {
+        throw IfConfigError.unhandledFunction(name: fnName, syntax: argExpr)
+      }
+
+      return actualVersion >= expectedVersion
+
+    case .canImport:
+      // Retrieve the first argument, which must not have a label. This is
+      // the module import path.
+      guard let firstArg = call.argumentList.first,
+        firstArg.label == nil
+      else {
+        throw IfConfigError.canImportMissingModule(syntax: ExprSyntax(call))
+      }
+
+      let importPath = firstArg.expression.trimmedDescription.split(separator: ".")
+      // FIXME: Check to make sure we have all identifiers here.
+
+      // If there is a second argument, it shall have the label _version or
+      // _underlyingVersion.
+      let version: CanImportVersion
+      if let secondArg = call.argumentList.dropFirst().first {
+        if secondArg.label?.text != "_version" && secondArg.label?.text != "_underlyingVersion" {
+          throw IfConfigError.canImportLabel(syntax: secondArg.expression)
+        }
+
+        let versionText: String
+        if let stringLiteral = secondArg.expression.as(StringLiteralExprSyntax.self),
+          stringLiteral.segments.count == 1,
+          let firstSegment = stringLiteral.segments.first,
+          case .stringSegment(let stringSegment) = firstSegment
+        {
+          versionText = stringSegment.content.text
+        } else {
+          versionText = secondArg.expression.trimmedDescription
+        }
+
+        guard let versionTuple = VersionTuple(parsing: versionText) else {
+          throw IfConfigError.invalidVersionOperand(name: "canImport", syntax: secondArg.expression)
+        }
+
+        // FIXME: Warning that the version can only have at most 4 components.
+
+        if secondArg.label?.text == "_version" {
+          version = .version(versionTuple)
+        } else {
+          assert(secondArg.label?.text == "_underlyingVersion")
+          version = .underlyingVersion(versionTuple)
+        }
+
+        if call.argumentList.count > 2 {
+          throw IfConfigError.canImportTwoParameters(syntax: ExprSyntax(call))
+        }
+      } else {
+        version = .unversioned
+      }
+
+      result = configuration.canImport(
+        importPath: importPath.map { String($0) },
+        version: version,
+        syntax: ExprSyntax(call)
+      )
+    }
+
+    // If we found a result, return it.
+    if let result { return result }
```

Comment:
We handle the `nil` case through a common point below, rather than unwrapping each time. It's likely the flow will change here as we think more about handling errors, so I'm not going to fret over it much now.